### ISSUE TITLE
Using corn loss with existing ordinal regression task.

### ISSUE THREAD

**Issue by sreenivasaupadhyaya** [ck37/coral-ordinal#25]
Hi @ck37 ,

Thanks for sharing this repo with Corn loss implementation. I see here that you have implemented both the corn layer as well as the CORN loss.

In my experiments, I already have a neural network trying to do ordinal regression (or classification) with N-1 output neurons (meaning i have N classes) and sigmoid activation on all the output neurons and the loss function is the sum of the individual Binary cross entropy loss from all the output neurons. Hence, I already have the ground truth outputs encoded in ordinal format.
I am using tensorflow 2.14.
Question:
1. Can I use the "loss = coral.OrdinalCrossEntropy()" in the place where i calculate the loss?
2. Does the "loss = coral.OrdinalCrossEntropy()" expect the ground truth outputs to be in integer labelled format (0 to N-1)?

Looking forward to your opinion.

**Comment by gmgeorg** [comment]
1. I think that should be possible, but I advise you use all the loss/activation/metric/layer functions from the coral-ordinal package; they are working together to make sure no mistakes are made from converting logits to probabilities, to cumulative probabilties, to eventual (predicted) labels.  
2. yes.
